corehq/apps/saved_reports/scheduled.py

```diff
@@ -95,14 +95,25 @@ def get_scheduled_report_ids(period, start_datetime=None, end_datetime=None):
             # Don't care if not on hour
             continue
         keys = _make_all_notification_view_keys(period, target_point_in_time)
+
         for key in keys:
             for result in ReportNotification.view(
                 "reportconfig/all_notifications",
                 reduce=False,
                 include_docs=False,
                 **key
             ).all():
-                yield result['id']
+                if period == 'hourly':
+                    hour = result['value'].get('hour')
+                    stop_hour = result['value'].get('stop_hour')
+
+                    # For backwards compatibility and a general safety measure
+                    if type(hour) != int or type(stop_hour) != int:
+                        yield result['id']
+                    elif hour <= target_point_in_time.hour <= stop_hour:
+                        yield result['id']
```

Comment:
instead of doing a type check you could try convert the values to `int` and handle exceptions:
```suggestion
                    try:
                        hour = int(result['value'].get('hour'))
                        stop_hour = int(result['value'].get('stop_hour'))
                        if hour <= target_point_in_time.hour <= stop_hour:
                            yield result['id']
                    except (TypeError, ValueError):
                        yield result['id']
```

Reply:
It would also be good to have a test for the error condition